src/pages/index.js

```diff
@@ -1,129 +1,218 @@
 import { FormValidator } from "../components/FormValidator";
 import "./index.css";
 import { Section } from "../components/Section.js";
-import { UserInfo } from "../components/UserInfo";
 import { PopupWithForm } from "../components/PopupWithForm.js";
 import Card from "../components/Card";
 import { PopupWithImage } from "../components/PopupWithImage";
+import { Api } from "../components/Api";
+import { UserInfo } from "../components/UserInfo";
+import { confirmationPopup } from "../components/ConfirmationPopup";
+import * as Constants from "../components/Constants";
 
-(function displayCards() {
-  // Elements
-  const imageClickHandler = ({ name, link }) => {
-    imagePopup.open({ name, link });
-  };
-  const cardTemplate =
-    document.querySelector("#card-template").content.firstElementChild;
-  const galleryList = document.querySelector("#gallery__list");
-
-  const items = [
-    {
-      name: "Yosemite Valley",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/yosemite.jpg",
-    },
-
-    {
-      name: "Lake Louise",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lake-louise.jpg",
-    },
-
-    {
-      name: "Bald Mountains",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/bald-mountains.jpg",
-    },
 
-    {
-      name: "Latemar",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/latemar.jpg",
+(function displayCards() {
+  const api = new Api({
+    baseUrl: "https://around-api.en.tripleten-services.com/v1",
+    headers: {
+      authorization: "9b29ae94-bb6f-470b-bccd-5f1bdf13a16a",
+      "Content-Type": "application/json",
     },
+  });
+  let initialCardsInfo = NaN
+
+  api
+    .getInitialcards()
+    .then((items) => {
+      initialCardsInfo = items.reverse();
+      const cardsSection = new Section(
+        {
+          items,
+          renderer: (cardData) => {
+            addCard(cardData.name, cardData.link, Constants.cardTemplate, cardData._id);
+          },
+        },
+        Constants.galleryList
+      );
+      cardsSection.renderItems();
+    })
+    .catch((error) => {
+      console.error("Caught an error: ", error.message);
+    })
+    
+    ;
+
+  const editAvatarPopup = new PopupWithForm("#avatar-modal", chageAvatar);
+
+  function chageAvatar(inputValue) {
+    api
```

Comment:
According to the task you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.

You don't do that for the avatar button